src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -316,8 +317,7 @@ FindNodeCheck(Node *node, bool (*check)(Node *))
 
 	if (IsA(node, RangeTblEntry))
 	{
-		/* query_tree_walker descends into RTEs */
-		return false;
+		return range_table_walker(list_make1(node), FindNodeCheck, check, 0);
```

Comment:
Hmz, `query_tree_walker` also calls `range_table_walker`, so this would create a large amount of redundant work.

Reply:
Not that much given that we return as soon as we find the node we're looking for? 
I'm ok with both implementations, just pick one 😄

Reply:
Yes, but it might take a while to find it, or maybe we won't find it. Can't we just pass a query to `ContainsDistributedTableRTE`?

Reply:
thinking about it a bit more, maybe we should have a different walker that does not invoke `query_tree_walker` and only considers rtable?